### ISSUE TITLE
Scikit-learn 1.3.0 release induce a bug 

### ISSUE THREAD

**Issue by AdrienTurchini** [scikit-learn-contrib/hdbscan#597]
`FAST_METRICS = KDTree.valid_metrics + BallTree.valid_metrics + ["cosine", "arccos"]` 
l.40 in https://github.com/scikit-learn-contrib/hdbscan/blob/master/hdbscan/hdbscan_.py 

doesn't work anymore with sklearn last release 1.3.0

**Comment by lmcinnes** [comment]
Yes, it looks like they've refactored how all of that works quite a bit. There's no easy way to handle that anymore looking at the current setup. I think the best options would be to hardcode a list of acceptable metrics from what KDTrees and BallTrees in sklearn currently accept. Eventually the dual-tree Boruvka code from here will be merged into sklearn proper, at which point this will be less of an issue. If you wanted to compile the metrics and put in a PR to fix this that would be appreciated. Otherwise it may have to wait until I get to it.

**Comment by lmcinnes** [comment]
This should be fixed in master now. I'll see if I can get a release out soon.

**Comment by AdrienTurchini** [comment]
Sorry, didn’t have time to work on that but thanks for the fix :)
